File: javascript/leetcode/toeplitz-matrix.js
/**
 * @param {number[][]} matrix
 * @return {boolean}
 */
var isToeplitzMatrix = function(matrix) {
    for (var i = 0; i < matrix.length - 1; i++) {
        var maxCheck = i === 0 ? matrix[0].length - 1 : 1;
        for (var j = 0; j < maxCheck; j++) {
            var val = matrix[i][j];
            // Check diagonals for this val
            var c = j + 1;
            var r = i + 1;
            while (r < matrix.length && c < matrix[0].length) {
                if (matrix[r][c] !== val) {
                    return false;
                }
                c++;
                r++;
            }
        }
    }
    return true;
};

File: java/leetcode/toeplitz-matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        for (int i = 0; i < matrix.length - 1; i++) {
            int maxCheck = i == 0 ? matrix[0].length - 1 : 1;
            for (int j = 0; j < maxCheck; j++) {
                int val = matrix[i][j];
                // Check diagonals for this val
                int c = j + 1;
                int r = i + 1;
                while (r < matrix.length && c < matrix[0].length) {
                    if (matrix[r][c] != val) {
                        return false;
                    }
                    c++;
                    r++;
                }
            }
        }
        return true;
    }
}
